core/station/impl/src/services/account.rs

```diff
@@ -390,38 +449,84 @@ impl AccountService {
 
         let mut balances = Vec::new();
         for mut account in accounts {
-            let balance_considered_fresh = match &account.balance {
-                Some(balance) => {
-                    let balance_age_ns = next_time() - balance.last_modification_timestamp;
-                    (balance_age_ns / 1_000_000) < ACCOUNT_BALANCE_FRESHNESS_IN_MS
-                }
-                None => false,
-            };
-            let balance: AccountBalance = match (&account.balance, balance_considered_fresh) {
-                (None, _) | (_, false) => {
-                    let blockchain_api =
-                        BlockchainApiFactory::build(&account.blockchain, &account.standard)?;
-                    let fetched_balance = blockchain_api.balance(&account).await?;
-                    let new_balance = AccountBalance {
-                        balance: candid::Nat(fetched_balance),
-                        last_modification_timestamp: next_time(),
-                    };
-
-                    account.balance = Some(new_balance.clone());
-
-                    self.account_repository
-                        .insert(account.to_key(), account.clone());
-
-                    new_balance
-                }
-                (Some(balance), _) => balance.to_owned(),
-            };
-
-            balances.push(AccountMapper::to_balance_dto(
-                balance,
-                account.decimals,
-                account.id,
-            ));
+            let account_key = account.to_key();
+            for account_asset in account.assets.iter_mut() {
```

Comment:
nit: deduplicate the list

Reply:
This is coming from the repository, no deduplication should be necessary.